File: 610A.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    unsigned long long int n;
    cin>>n;

    long long int x,count, a, i=1, ans=0;
    x=n/2;
    count=n/4;

    if(n%2!=0) cout<<"0"<<endl;
    else{
        while(count--){
            a=x-i;

            if(a!=i) ans++;
            i++;
        }

        cout<<ans<<endl;
    }
}

File: 610A.py
n = int(input())

x = n//2
count = n//4

i = 1
ans = 0

if n%2 != 0:
    print(ans)
else:
    while count > 0:
        a = x - i

        if a != i:
            ans += 1

        i += 1
        count -= 1

    print(ans)